pkg/synthetictests/duplicated_events.go

```diff
@@ -147,3 +214,85 @@ func getTimesAnEventHappened(message string) (string, int) {
 	}
 	return matches[0][1], int(times)
 }
+
+// isConsoleReadinessDuringInstallation returns true if the event is for console readiness and it happens during the
+// initial installation of the cluster.
+// we're looking for something like
+// > ns/openshift-console pod/console-7c6f797fd9-5m94j node/ip-10-0-158-106.us-west-2.compute.internal - reason/ProbeError Readiness probe error: Get "https://10.129.0.49:8443/health": dial tcp 10.129.0.49:8443: connect: connection refused
+// with a firstTimestamp before the cluster completed the initial installation
+func isConsoleReadinessDuringInstallation(monitorEvent monitorapi.EventInterval, kubeClientConfig *rest.Config) bool {
+	if !strings.Contains(monitorEvent.Locator, "ns/openshift-console") {
+		return false
+	}
+	if !strings.HasPrefix(monitorEvent.Locator, "pod/console-") {
+		return false
+	}
+	if !strings.Contains(monitorEvent.Locator, "Readiness probe") {
+		return false
+	}
+	if !strings.Contains(monitorEvent.Locator, "connect: connection refused") {
+		return false
+	}
+	tokens := strings.Split(monitorEvent.Locator, " ")
+	tokens = strings.Split(tokens[1], "/")
+	podName := tokens[1]
+
+	if kubeClientConfig == nil {
+		// default to OK
+		return true
+	}
+
+	// This block gets the time when the cluster completed installation
+	configClient, err := configclient.NewForConfig(kubeClientConfig)
+	if err != nil {
+		// default to OK
+		return true
+	}
+	clusterVersion, err := configClient.ConfigV1().ClusterVersions().Get(context.TODO(), "version", metav1.GetOptions{})
+	if err != nil {
+		// default to OK
+		return true
+	}
+	if len(clusterVersion.Status.History) == 0 {
+		// default to OK
+		return true
+	}
+	initialInstallHistory := clusterVersion.Status.History[len(clusterVersion.Status.History)-1]
+	if initialInstallHistory.CompletionTime == nil {
+		// default to OK
+		return true
+	}
+
+	// this block gets the actual event from the API.  This is ugly, but necessary because we don't have real events.
+	// It may be interesting to track real events, but it would be expensive.
+	kubeClient, err := kubernetes.NewForConfig(kubeClientConfig)
+	if err != nil {
+		// default to OK
+		return true
+	}
+	consoleEvents, err := kubeClient.CoreV1().Events("openshift-console").List(context.TODO(), metav1.ListOptions{})
```

Comment:
we are going to query this list for each event?
is this ok or can be overwhelming?

Reply:
> we are going to query this list for each event?
> is this ok or can be overwhelming?

Good question.  Since we decimate the  list before, we only do this for readiness probe failures for console pods in their namespace.  About 8 or so of these usually.